ambry-messageformat/src/test/java/com.github.ambry.messageformat/MessageFormatSendTest.java

```diff
@@ -137,26 +162,37 @@ private void doSendWriteSingleMessageTest(ByteBuffer encryptionKey, ByteBuffer e
         new BlobProperties(blob.length, serviceId, ownerId, contentType, false, 100, accountId, containerId,
             encryptionKey != null);
     MessageFormatInputStream putStream;
+    MessageFormatRecord.MessageHeader_Format header;
     if (putFormat.equals(PutMessageFormatInputStream.class.getSimpleName())) {
       putStream = new PutMessageFormatInputStream(storeKey, encryptionKey, properties, ByteBuffer.wrap(userMetadata),
           new ByteBufferInputStream(ByteBuffer.wrap(blob)), blob.length, BlobType.DataBlob);
+      if (encryptionKey != null) {
```

Comment:
in order to not cause any unrelated problems, use `ByteBuffer.duplicate()` and use that in the `PutMessageFormatInputStream()` that you use to get the header.